lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -350,6 +353,9 @@ export default function getAPIImplementation(): APIInterface {
                   user: User;
                   email: string;
               }
+            | {
+                  status: "USER_BANNED_ERROR";
+              }
```

Comment:
no need to modify this API, since the token cant be generated in the first place in case the user is banned

Reply:
Instead, modify the recipe interface.